packages/@aws-cdk/assets/lib/asset.ts

```diff
@@ -40,11 +40,29 @@ export interface AssetProps {
   readonly readers?: iam.IGrantable[];
 }
 
+export interface IAsset extends cdk.IConstruct {
+  /**
+   * A hash of the source of this asset, which is available at construction time. As this is a plain
+   * string, it can be used in `Construct` ids in order to enforce creation of a new resource when
+   * the content hash has changed.
+   */
+  readonly sourceHash: string;
+
+  /**
+   * A hash of the bundled for of this asset, which is only available at deployment time. As this is
+   * a late-bound token, it may not be used in `Construct` ids, but can be passed as a resource
+   * property in order to force a change on a resource when an asset is effectively updated. This is
+   * more reliable than `sourceHash` in particular for assets which bundling phase involve external
+   * resources that can change over time (such as Docker image builds).
+   */
+  readonly bundleHash: string;
```

Comment:
I think this should be called `artifactHash`. A bit more general than bundle which doesn't really apply to Docker image or to a single non-zipped files, and the terminology `source => artifact` is quite clear